pkg/types/installconfig.go

```diff
@@ -446,6 +450,29 @@ type ClusterNetworkEntry struct {
 	DeprecatedHostSubnetLength int32 `json:"hostSubnetLength,omitempty"`
 }
 
+// OVNKubernetesConfig configures the ovn-kubernetes sdn plugin.
+type OVNKubernetesConfig struct {
+	// ipv4 allows users to configure IP settings for IPv4 connections. When omitted,
+	// this means no opinions and the default configuration is used. Check individual
+	// fields within ipv4 for details of default values.
+	// +optional
+	IPv4 *IPv4OVNKubernetesConfig `json:"ipv4,omitempty"`
```

Comment:
Nit: Can we call this `IPv4Settings`/`IPV4Config` ? Just `IPv4`  has a special meaning. This is a personal preference and there might be precedence for the kind of usage within Installer or other OpenShift components.

Reply:
>  there might be precedence

This API is indeed copied directly from o/api. 

> Just IPv4 has a special meaning. 

I'm not sure if this is what you meant, but I agree that it would be more readable to include "config" (of some sort) when using this. But I do think we should keep the naming of the fields as is, because the parent is `OVNKubernetesConfig` and I think `OVNKubernetesConfig.IPv4` is pretty clear, and avoids stuttering like `OVNKubernetesConfig.IPv4Config`. 

I think the problem is my variable naming in code. I do a lot of `ovn.IPv4.InternalJoinSubnet` in the code, so I will update that to `ovnCfg.IPv4.InternalJoinSubnet`. WDYT?

Reply:
> I'm not sure if this is what you meant, but I agree that it would be more readable to include "config" (of some sort) when using this. But I do think we should keep the naming of the fields as is, because the parent is `OVNKubernetesConfig` and I think `OVNKubernetesConfig.IPv4` is pretty clear, and avoids stuttering like `OVNKubernetesConfig.IPv4Config`.
> 
> I think the problem is my variable naming in code. I do a lot of `ovn.IPv4.InternalJoinSubnet` in the code, so I will update that to `ovnCfg.IPv4.InternalJoinSubnet`. WDYT?

I agree `OVNKubernetesConfig.IPv4Config` is not ideal. I do not want to bikeshed.
Maybe  `OVNKubernetesConfig.IPV4InternalJoinSubnet` would have been sufficient (without having an intermediate `IPv4` category) . Can grow to include other config like `OVNKubernetesConfig.IPV6InternalJoinSubnet`. But this is making assumptions about the future of this configuration.